Shared/DsaController.cpp

```diff
@@ -86,14 +88,24 @@ DsaController::DsaController(QObject* parent):
 {
   // setup config settings
   setupConfig();
-  m_scene->setInitialViewpoint(defaultViewpoint());
+  m_scene->setInitialViewpoint(viewpointFromJson(defaultViewpoint()));
   m_dataPath = m_dsaSettings["RootDataDirectory"].toString();
 
   connect(m_scene, &Scene::errorOccurred, this, &DsaController::onError);
 
-  connect(ToolResourceProvider::instance(), &ToolResourceProvider::sceneChanged, this, [this]()
+  connect(ToolResourceProvider::instance(), &ToolResourceProvider::sceneChanged, this, [this, firstLoad{true}]() mutable
   {
     m_scene = ToolResourceProvider::instance()->scene();
+
+    // Here we set up the initial location of the scene fron the config. We do this on the first scene
+    // we load when the application starts up. On any subsequent scenes that the user loads
+    // via the UI, we take this new initial location as the source of truth from then on and
+    // write it to file.
+    if (!m_scene)
+      return;
+
+    updateInitialLocationOnSceneChange(firstLoad);
+    firstLoad = false;
```

Comment:
I don't see where we use `firstLoad` after this call?

Reply:
Only in this call do we use it, it's only of interest whenever the scene changes.